subprojects/gradle/instrumentation-tests/src/main/kotlin/com/avito/instrumentation/reservation/devices/provider/MockDevicesProvider.kt

```diff
@@ -0,0 +1,98 @@
+package com.avito.instrumentation.reservation.devices.provider
+
+import com.avito.instrumentation.reservation.request.Device.MockEmulator
+import com.avito.instrumentation.reservation.request.Reservation
+import com.avito.runner.service.model.TestCaseRun
+import com.avito.runner.service.worker.device.Device
+import com.avito.runner.service.worker.device.DeviceCoordinate
+import com.avito.runner.test.mock.MockActionResult
+import com.avito.runner.test.mock.MockDevice
+import com.avito.utils.logging.CILogger
+import com.avito.utils.logging.commonLogger
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.channels.Channel
+import kotlinx.coroutines.channels.ReceiveChannel
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.launch
+import org.funktionale.tries.Try
+import java.util.concurrent.TimeUnit
+
+class MockDevicesProvider(private val logger: CILogger) : DevicesProvider {
+
+    private val devices = Channel<Device>(Channel.UNLIMITED)
+
+    @ExperimentalCoroutinesApi
+    override fun provideFor(reservations: Collection<Reservation.Data>, scope: CoroutineScope): ReceiveChannel<Device> {
+        val devicesRequired = reservations.fold(0, { acc, reservation -> acc + reservation.count })
+        scope.launch(Dispatchers.IO) {
```

Comment:
constructor

Reply:
won't do in this PR scope
also needed in LocalDevicesProvider; better to change simultaneously